deployment/common/service.yaml

```diff
@@ -4,15 +4,24 @@ metadata:
  name: contour
  namespace: heptio-contour
  annotations:
-  # This annotation puts the AWS ELB into "HTTP" mode so that it adds the
-  # `X-Forwarded-For` header on the request. If you aren't running on AWS this
-  # should be a no-op.
-  service.beta.kubernetes.io/aws-load-balancer-backend-protocol: http
+  # This annotation puts the AWS ELB into "TCP" mode so that it does not
+  # to http negotiation at the ELB edge. This means the remote address of
+  # all connectinos will appear to be the internal address of the ELB.
+  service.beta.kubernetes.io/aws-load-balancer-backend-protocol: tcp
+  # To recover the remote ip 
+  # 1. Add the --use-proxy-protocol flag to your deployment or daemonset's contour serve command.
+  # 2. uncomment this line to enable PROXY protocol # on the ELB
+  # service.beta.kubernetes.io/aws-load-balancer-proxy-protocol: "*"
```

Comment:
Not sure where to put this comment, so duplicating in what seem to be the two obvious places.
Edit first comment (starting at l7):

# This annotation configures the AWS ELB for TLS passthrough.
# This means that the remote address of each connection appears to be
# the internal address of the ELB.
# To recover the external address:
# 1. Add the --use-proxy-protocol flag to the command to retrieve the address.
# 2. Uncomment the following line to enable the proxy protocol on the ELB.

Do we also want to add a comment line after the # service.beta ... line to point them to the docs for more info? Or is that getting too carried away?

Reply:
Having a docs page on the proxy protocol makes sense.  This is enough of a topic we probably want to delve in a bit and have pointers to the ELB docs and such.

Also worth calling out that this isn't needed if NLB is used.  That is a new feature in 1.9.

Reply:
We could also have another version of this service (aws-service.yaml) and another set of links to pull it all together.  At the end of the day this is screaming for ksonnet. :/

Reply:
I think pulling this out into a proxy-protocol.md is the best solution. Unlike the elb mode, it won't be transparent on other clusters.